.taskcluster.yml

```diff
@@ -2,132 +2,117 @@
-    extra:
-      github:
-        events:
-          - release
-    scopes:
-      - "secrets:get:project/application-services/publish"
-    payload:
-      maxRunTime: 7200
-      image: 'mozillamobile/rust-component:buildtools-27.0.3-ndk-r15c-ndk-version-21-rust-stable-1.28.0-rust-beta-1.29.0-beta.15'
-      command:
-        - /bin/bash
-        - '-c'
-        - >-
-          export TERM=dumb
-          && git clone '{{ event.head.repo.url }}' application-services
-          && cd application-services
-          && git config advice.detachedHead false
-          && git checkout '{{ event.version }}'
-          && python automation/taskcluster/release/fetch-bintray-api-key.py
-          && ./scripts/taskcluster-android.sh
-          && ./gradlew --no-daemon clean :fxa-client-library:assembleRelease :logins-library:assembleRelease
-          && ./gradlew bintrayUpload --debug -PvcsTag="{{ event.head.sha }}"
-      artifacts:
-        'public/bin/mozilla/fxaclient-release-{{ event.version }}.aar':
-          type: 'file'
-          path: '/build/application-services/fxa-client/sdks/android/library/build/outputs/aar/fxaclient-release.aar'
-        'public/bin/mozilla/logins-release-{{ event.version }}.aar':
-          type: 'file'
-          path: '/build/application-services/logins-api/android/library/build/outputs/aar/logins-release.aar'
-      features:
-        taskclusterProxy: true
-    routes:
-      - index.project.fxaclient.builds.{{ event.version }}
-    metadata:
-      name: fxa-client Android Build - Release ({{ event.version }})
-      description: Builds the FxA client and the Logins API for Android architectures.
-      owner: '{{ event.head.user.email }}'
-      source: '{{ event.head.repo.url }}'
+  - $if: 'tasks_for == "github-pull-request"'
+    then:
+      $if: 'event.action in ["opened", "reopened", "synchronize"]'
+      then:
+        taskGroupId: {$eval: as_slugid("decision_task")}
+        taskId: {$eval: as_slugid("decision_task")}
+        provisionerId: aws-provisioner-v1
+        workerType: application-services-r
+        created: {$fromNow: ''}
+        deadline: {$fromNow: '1 day'}
+        metadata:
+          name: "Application Services: GitHub decision task"
+          description: ""
+          owner: &task_owner ${event.pull_request.head.user.login}@users.noreply.github.com
+          source: &task_source ${event.pull_request.head.repo.url}
+        scopes:
+          - "queue:scheduler-id:taskcluster-github"
+          # So that we can use our own "application-services-r" worker-type.
+          - "queue:create-task:highest:aws-provisioner-v1/application-services-r"
+
+          # So that we can cache task outputs for re-use.
+          - "queue:route:index.project.application-services.*"
+
+          # So that we can re-use Gradle/Cargo/sccache bits between tasks.
+          - "docker-worker:cache:application-services-*"
+
+        payload:
+          maxRunTime: {$eval: '20 * 60'}
```

Comment:
Weird. Is there a reason to specify this separately?

Reply:
The repetition of blocks based on the different events?  Not really, I just couldn't be bothered to smooth it out.  Filed https://github.com/mozilla/application-services/issues/340 to improve that.

Reply:
I more meant using some sort of evaluation syntax to compute 20 * 60